### DISCUSSION TITLE
[FEAT] Rename Disks

### DISCUSSION THREAD

**Discussion by Mike710Shine** [AnalogJ/scrutiny#926] [Feature Requests, Ideas]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
I would like to be able to name the Disks because i have 5 Data Disks and 1 Paraty and when i look in scrutiny i have to look ot sdg or sda or so to find the right disk. It would be eseaser when i can name the disks.

**Additional context**
For example screenshot edited.
![grafik](https://github.com/AnalogJ/scrutiny/assets/19763591/65bc5c8d-595b-40ca-8282-e685dab96699)

**Comment by HotBlack77** [comment]
This would be really useful. 
Could it be as simple as letting users add a list of UUIDs and names in the config file?

**Comment by AnalogJ** [comment]
Scrutiny already collects your device partition labels. Have you tried changing the `Display title` setting?

<img width="645" alt="Screen Shot 2024-01-24 at 6 32 05 AM" src="https://github.com/AnalogJ/scrutiny/assets/891875/2f7ff0b1-4def-4e57-8ede-b1e5f16bd3e9">

**Comment by Mike710Shine** [comment]
Yes i tried it. But more than SDF i didn't get. Because i didn't have set an patition name. I would recommend it to change the name of each Harddrive. This would be nice.

**Comment by HotBlack77** [comment]
I get the following using "display title" option in settings (the one you showed in your screenshot). 
 - Name = device path, e.g. /dev/sdX (although its styled in all-caps) 
 - Serial ID = device serial, e.g. /BY-ID/ATA-TOSHIBA_HDWQ140_xxxxx
 - UUID = Same as for Name option, that is device path
 - Label = Same as for Name option, that is device path

I am running master#a3dfce3 in a docker container using the compose file you suggest.

**Comment by xHyperElectric** [comment]
This is my most desired feature

**Comment by sherbibv** [comment]
I too am interested in this feature. I have multiple drives with the same name/UUID/Label with the current implementation and I can't tell them appart. Being able to add a label in the webui would be great

**Comment by AnalogJ** [comment]
I think we can support overriding the disk label in the collector config file by making changes to https://github.com/AnalogJ/scrutiny/blob/master/collector/pkg/models/scan_override.go

and then wiring it up to the Label field in the device. Anyone have some time to take a look and open a PR?

**Comment by Eddy555** [comment]
I've found on my system any disk with a partition does not show the label
i.e. the following just shows /dev/sda
/dev/sda
/dev/sda1  Label = Disk 1

However I have 1 disk that is
/dev/sdb Label = Disk 2
and this works fine,

**Comment by robflate** [comment]
Has any progress been made on this? Thanks.

**Comment by kaysond** [comment]
This issue is being closed automatically as a part of an effort to streamline and prioritize development effort. If it is still relevant, please reply to this comment and we will convert it to a discussion.

**Comment by Eddy555** [comment]
Hi, this is still relevant for me, as outlined in my earlier comment. Thanks

**Reply by kaysond** [reply]
I'm thinking it makes more sense to implement this on the web/backend side. Once #923 is addressed, we'll have a hopefully truly unique way to identify drives and don't have to rely on device paths. Having it on the backend also means that the drive could be moved to another host (e.g. USB drive) and still show up with the right label without needing to change the a collector config.

Thoughts?

**Reply by newhinton** [reply]
Since #923 got merged, do you need/want help with making this happen?

I'd like to see this too, especially in notifications. Only having the drive path and uuid is not that descriptive :D

**Reply by kaysond** [reply]
> Since #923 got merged, do you need/want help with making this happen?
> 
> I'd like to see this too, especially in notifications. Only having the drive path and uuid is not that descriptive :D

Some help would be great! Feel free to open a draft PR with a spec for how you'd implement it on the backend and frontend and we can discuss.

Some random thoughts: we need a migration to add a `friendly_name` column to the devices table. The front end currently shows the host name both as section headers and before the drive name, which is redundant. I was going to remove that in this patch. Most of the Settings -> Display Title don't seem to work for me. Might be worth cleaning that up at the same time (and because we should have a friendly name option. or maybe just use name? these are things that need to be fleshed out). The drive name link brings up the drive details page, so I think the best thing to do would be add a Set Friendly Name option to the 3-dots menu. We'll also need a new endpoint in the backend that takes the scrutiny uuid and friendly name and sets the latter for the former.

Thanks!

**Reply by newhinton** [reply]
Sure!

I started working on this, but i cant seem to make changes to the frontend. I followed the CONTRIBUTING.MD, and i run the following in parallel:

```
go run webapp/backend/cmd/scrutiny/scrutiny.go start --config ./scrutiny.yaml
npm run build:prod -- --watch --output-path=../../dist
```

I get a "working" scrutiny instance up, (as in, if i add the demodata it shows up), but if i make changes to the frontend, nothing happens, even after restarting both.

`npm run start -- --serve-path="/web/" --port 4200` does not work either, it is just stuck at the initial loading-screen, and `http://localhost:4200/api/summary` 404's.

Any tips?

**Reply by kaysond** [reply]
Maybe try running it in the dev container? It's suspicious that even after restarting both you don't see the updates.

> `npm run start -- --serve-path="/web/" --port 4200` does not work either, it is just stuck at the initial loading-screen, and `http://localhost:4200/api/summary` 404's.

When you do `npm run start`, it only serves the frontend with mock data built into the frontend. The API routes are served by the backend, so you won't have those.

Also - are you running the influxdb container? That could explain the issues with running both.

**Reply by newhinton** [reply]
Okay got it.

How should i provide the device name? The frontend seems to not allow to set device-specific settings, which makes sense.
One Idea i had was to provide it via ENV, something like `NAME_Scrutiny-UUID=my cool friendly name`. Any other suggestions?

**Comment by newhinton** [comment]
I started work on this in https://github.com/AnalogJ/scrutiny/pull/975

**Comment by kaysond** [comment]
lets move discussion to #911.

this is on my list to implement after a few remaining bugs are squashed
